.devcontainer/README.md

```diff
@@ -0,0 +1,204 @@
+  - Install [Dev Container Extension Pack](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) via the command line
+ ```bash
+ code --install-extension ms-vscode-remote.remote-containers
+ ```
+
+### Steps
+
+1. Clone the Repository
+ ```bash
+ git clone git@github.com:nvidia-holoscan/holohub.git
+ ```
+2. Open the cloned directory in the terminal.
+
+3. Launch a Dev Container with the `./dev_container` script as follows:
+
+ ```bash
+ ./dev_container vscode
+ ```
+ The above command starts a new Dev Container for Holohub using the default [Dockerfile](../Dockerfile).
+
+
+4. VS Code will build and initialize the selected Dev Container. This can take a few minutes the first time.
+
+5. Once initialized, a new VS Code window will open with the following prompts. Click **Yes** and **Trust Folder & Continue** to continue the Dev Container build process.
+
+![VS Code External Application](./static/vscode-prompt-ext-app.png)
+![VS Code Trust Directory](./static/vscode-prompt-trust.png)
+
+6. When ready, the Holohub directory is mirrored into the container under `/workspace/holohub` to preserve any changes.
+
+
+### Debugging Holohub Applications
+
+Most Holohub applications are pre-configured with one or more launch profiles. Click the **Run & Debug** tab and select the application you want to run/debug from the dropdown.
+
+#### Debugging Multi-Fragment Applications
+
+To debug multi-fragment applications, find and locate launch profiles prefixed with `(compound)`.
+
+For example, the [Distributed Endoscopy Tool Tracking](../applications/endoscopy_tool_tracking_distributed/) application is configured with `(compound) endoscopy_tool_tracking_distributed/cpp` and `(compound) endoscopy_tool_tracking_distributed/python` launch profiles.
+These launch profiles start three debugging sessions, one for each fragment (`video_in`, `inference`, and `viz`), and enable debugging of all three processes at once.
+
+> 💡 Note: Launch profiles prefixed with `(compound)` for Python applications default to `debugpy` debgguer which only allows debugging of Python code.  To debug both Python and C++ code, modify the  compound launch profile in the [launch.json](../.vscode/launch.json) and change `(debugpy)` to `(pythoncpp)`.
+
+
+#### Step into Holoscan Source Code
+
+The Holohub Dev Container derives from the [Holoscan NGC Container](https://catalog.ngc.nvidia.com/orgs/nvidia/teams/clara-holoscan/containers/holoscan), which supports debugging of Holoscan source code in C++ and Python. Add the Holoscan source code to the workspace to simplify the process by clicking the **Add Folder to Workspace** menu item from the **File** menu. Enter `/workspace/holoscan-sdk/` in the **Add Folder to Workspace** dialog box. If VS Code prompts you to reload the window, please do so. Expect to see both the *holohub* and *holoscan-sdk* folders appear under the Explorer tab.
```

Comment:
In my case the holoscan-sdk folder is not available once I am in the vscode dev_container. I also tried mounting it when starting the vscode container, e.g. 
`./dev_container vscode --docker_opts "-v </host/holoscan-sdk>:/worksapce/holoscan-sdk" `

Can you clarify how you made both available in the devcontainer?

Reply:
Hi @maximilianofir, in order to add Holoscan SDK source code, you will need Holoscan container v2.3 or higher, as noted in the readme. Thanks.